src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -33,122 +34,126 @@ public override WebSocketState State
+                            st = WebSocketState.CloseSent;
+                        }
+                    }
+                    return FastState = st;
+                } // lock
             }
         }
 
         private WebSocketState FastState
         {
             get
             {
-#if FEATURE_WASM_THREADS
-                lock (_thisLock)
+                lock (_lockObject)
                 {
-#endif
                     return _state;
-#if FEATURE_WASM_THREADS
-                } //lock
-#endif
+                } // lock
             }
             set
             {
-#if FEATURE_WASM_THREADS
-                lock (_thisLock)
+                lock (_lockObject)
                 {
-#endif
                     _state = value;
-#if FEATURE_WASM_THREADS
-                } //lock
-#endif
+                } // lock
             }
         }
 
-        public override WebSocketCloseStatus? CloseStatus => _closeStatus;
-        public override string? CloseStatusDescription => _closeStatusDescription;
-        public override string? SubProtocol
+        public override WebSocketCloseStatus? CloseStatus
         {
             get
             {
-#if FEATURE_WASM_THREADS
-                lock (_thisLock)
+                lock (_lockObject)
                 {
-#endif
-                    ThrowIfDisposed();
-                    if (_innerWebSocket == null) throw new InvalidOperationException(SR.net_WebSockets_NotConnected);
-#if FEATURE_WASM_THREADS
-                } //lock
-#endif
+                    if (_closeStatus != null)
+                    {
+                        return _closeStatus;
+                    }
+                    if (_disposed || _aborted || _cancelled)
+                    {
+                        return null;
+                    }
+                    return GetCloseStatus();
+                }
+            }
+        }
 
-#if FEATURE_WASM_THREADS
-                return _innerWebSocket.SynchronizationContext.Send(BrowserInterop.GetProtocol, _innerWebSocket);
-#else
-                return BrowserInterop.GetProtocol(_innerWebSocket);
-#endif
+        public override string? CloseStatusDescription
+        {
+            get
+            {
+                lock (_lockObject)
+                {
+                    if (_closeStatusDescription != null)
+                    {
+                        return _closeStatusDescription;
+                    }
+                    if (_disposed || _aborted || _cancelled)
+                    {
+                        return null;
+                    }
+                    return GetCloseStatusDescription();
+                }
+            }
+        }
 
+        public override string? SubProtocol
+        {
+            get
+            {
+                ThrowIfDisposed();
+                lock (_lockObject)
+                {
+                    if (_innerWebSocket == null) throw new InvalidOperationException(SR.net_WebSockets_NotConnected);
```

Comment:
The `BrowserInterop.GetProtocol` is returning `null` when web socket null or disposed, shouldn't we use the same behavior here as well?

Reply:
this is pre-existing behavior, I'll not change it as part of this PR

Reply:
I don't understand. You are changing it, don't you?

Reply:
no, I'm not. This is the old one

![image](https://github.com/dotnet/runtime/assets/271576/3ffec89d-4047-4e48-908d-9bdf952e4838)